main.go

```diff
@@ -59,10 +114,52 @@ func main() {
 	// Output the generated review
 	fmt.Println("Generated Review:")
 	fmt.Println(review)
+	for _, comment := range reviewComments {
+		fmt.Printf("File: %s, Line: %d\nComment: %s\n", *comment.Path, *comment.Position, *comment.Body)
+	}
 
-	// Post the review if a flag is set
-	if *approve || *requestChanges {
-		err = postReview(client, ctx, *owner, *repo, *prNumber, review, *approve)
+	if *dryRun {
+		// If dry run, don't post the review, just output the results
+		fmt.Println("Dry run: Review not posted to GitHub.")
+		return
+	}
+
+	// Check if the reviewer is the PR author
+	isSelfReview := user.GetLogin() == pr.User.GetLogin()
+
+	if isSelfReview {
+		// Post the review as a comment instead
+		commentBody := review
+		if action == "approve" {
+			commentBody += "\n\n**Note:** This is a self-approved PR."
+		} else if action == "request_changes" {
+			commentBody += "\n\n**Note:** This is a self-requested change."
+		}
+
+		comment := &github.IssueComment{
+			Body: github.String(commentBody),
+		}
+
+		_, _, err := client.Issues.CreateComment(ctx, *owner, *repo, *prNumber, comment)
```

Comment:
"This comment does not clarify why the self-review is displayed. A precise mention regarding the situation could enhance clarity for readers.